File: pos-scanner-web/backend/models/mobile_device.py
# backend/models/mobile_device.py
from extensions import db
from datetime import datetime


class MobileDevice(db.Model):
    """移动设备模型"""
    __tablename__ = 'mobile_devices'

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(200), nullable=False)  # 设备名称
    device_type = db.Column(db.String(100))  # 设备类型
    image_a = db.Column(db.String(500))  # A面图片路径
    image_b = db.Column(db.String(500))  # B面图片路径
    system_version = db.Column(db.String(100))  # 系统版本
    created_at = db.Column(db.DateTime, default=datetime.now)
    updated_at = db.Column(db.DateTime, default=datetime.now, onupdate=datetime.now)

    # 占用相关
    occupier_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    purpose = db.Column(db.String(500))  # 用途
    start_time = db.Column(db.DateTime)  # 占用开始时间
    end_time = db.Column(db.DateTime)  # 占用结束时间（释放时间）

    # 关联用户
    occupier = db.relationship('User', backref='occupied_devices')

    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'deviceType': self.device_type,
            'imageA': self.image_a,
            'imageB': self.image_b,
            'systemVersion': self.system_version,
            'createdAt': self.created_at.isoformat() if self.created_at else None,
            'updatedAt': self.updated_at.isoformat() if self.updated_at else None,
            'isOccupied': self.end_time is not None and self.end_time > datetime.now(),
            'occupier': self.occupier.username if self.occupier else None,
            'occupierId': self.occupier_id,
            'purpose': self.purpose,
            'startTime': self.start_time.isoformat() if self.start_time else None,
            'endTime': self.end_time.isoformat() if self.end_time else None
        }

File: pos-scanner-web/backend-go/internal/models/mobile_device.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type MobileDevice struct {
	ID            uint           `gorm:"primaryKey" json:"id"`
	Name          string         `gorm:"size:200;not null" json:"name"`
	DeviceType    *string        `gorm:"size:100" json:"device_type"`
	SN            *string        `gorm:"size:100" json:"sn"`
	ImageA        *string        `gorm:"size:500" json:"image_a"`
	ImageB        *string        `gorm:"size:500" json:"image_b"`
	SystemVersion *string        `gorm:"size:100" json:"system_version"`
	CreatedAt     time.Time      `json:"created_at"`
	UpdatedAt     time.Time      `json:"updated_at"`
	OccupierID    *uint          `gorm:"index" json:"occupier_id"`
	Purpose       *string        `gorm:"size:500" json:"purpose"`
	StartTime     *time.Time     `json:"start_time"`
	EndTime       *time.Time     `json:"end_time"`
	Occupier      *User          `gorm:"foreignKey:OccupierID" json:"-"`
	DeletedAt     gorm.DeletedAt `gorm:"index" json:"-"`
}

func (MobileDevice) TableName() string {
	return "mobile_devices"
}

func (m *MobileDevice) ToDict() map[string]interface{} {
	isOccupied := false
	if m.EndTime != nil && m.EndTime.After(time.Now()) {
		isOccupied = true
	}

	occupier := ""
	if m.Occupier != nil {
		if m.Occupier.Name != nil && *m.Occupier.Name != "" {
			occupier = *m.Occupier.Name
		} else {
			occupier = m.Occupier.Username
		}
	}

	result := map[string]interface{}{
		"id":            m.ID,
		"name":          m.Name,
		"deviceType":    m.DeviceType,
		"sn":            m.SN,
		"imageA":        m.ImageA,
		"imageB":        m.ImageB,
		"systemVersion": m.SystemVersion,
		"createdAt":     m.CreatedAt.Format(time.RFC3339),
		"updatedAt":     m.UpdatedAt.Format(time.RFC3339),
		"isOccupied":    isOccupied,
		"occupier":      occupier,
		"occupierId":    m.OccupierID,
		"purpose":       m.Purpose,
	}

	if m.StartTime != nil {
		result["startTime"] = m.StartTime.Format(time.RFC3339)
	} else {
		result["startTime"] = nil
	}

	if m.EndTime != nil {
		result["endTime"] = m.EndTime.Format(time.RFC3339)
	} else {
		result["endTime"] = nil
	}

	return result
}
